sdk/search/Azure.Search/src/SearchFilter.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Search
+{
+    /// <summary>
+    /// The SearchFilter class is used to help construct valid OData filter
+    /// expressions, like the kind used by <see cref="SearchOptions.Filter"/>,
+    /// by automatically replacing, quoting, and escaping interpolated
+    /// parameters.
+    /// <see href="https://docs.microsoft.com/en-us/azure/search/search-filters"/>
+    /// </summary>
+    public static class SearchFilter
+    {
+        /// <summary>
+        /// Create an OData filter expression from an interpolated string.  The
+        /// interpolated values will be quoted and escaped as necessary.
+        /// </summary>
+        /// <param name="filter">An interpolated filter string.</param>
+        /// <returns>A valid OData filter expression.</returns>
+        public static string Create(FormattableString filter) =>
+            Create(filter, null);
+
+        /// <summary>
+        /// Create an OData filter expression from an interpolated string.  The
+        /// interpolated values will be quoted and escaped as necessary.
+        /// </summary>
+        /// <param name="filter">An interpolated filter string.</param>
+        /// <param name="formatProvider">
+        /// Format provider used to convert values to strings.
+        /// <see cref="CultureInfo.InvariantCulture"/> is used as a default.
+        /// </param>
+        /// <returns>A valid OData filter expression.</returns>
+        public static string Create(FormattableString filter, IFormatProvider formatProvider)
```

Comment:
Do yuou need a CreateRaw overload that allows passing pre-formatted stirng?

Reply:
Do you mean for users who don't want us to do any formatting?  They could just skip this type and assign it directly to `SearchOptions.Filter`, etc.  Or are you talking about something else?

Reply:
> Do you mean for users who don't want us to do any formatting

Yes.

Reply:
Yeah - in that case you'd just skip this type entirely and assign your already formatted string to `*Options.Filter`.